data/transactions/transaction.go

```diff
@@ -161,6 +161,7 @@ type TxGroup struct {
 	// together, sequentially, in a block in order for the group to be
 	// valid.  Each hash in the list is a hash of a transaction with
 	// the `Group` field omitted.
+	// These are all `Txid` which is equivalent to `crypto.Digest`
```

Comment:
could you please add a test into `transaction_test.go` checking `tx.ID() == crypto.HashObj(tx)` ? In code I see they are the same but this is not enforced.

Reply:
agreed, would be good to put that assertion in this PR